tensorboard/components/tensor_widget/types.ts

```diff
@@ -0,0 +1,211 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {BaseTensorHealthPill} from "./health-pill-types";
+
+/**
+ * The specs for deferred view into a tensor.
+ *
+ * A tensor is a typed, multi-dimensional array.
+ * This interface abstracts away the backing storage of the tensor value.
+ * It allows on-demand retrieval into any element or sub-array of the tensor.
+ */
+export interface TensorView {
```

Comment:
I renamed the parameter to `tensorView`. 
Briefly, the reason for using a view is so that we can handle cases in which the tensor itself is too large to be held in its entirety in the browser memory. This is one of the key considerations of TensorWidget's design. See the design doc for more info.

Reply:
Sure. Revised the doc string accordingly.